### ISSUE TITLE
nvim-cmp with border issue disucussion

### ISSUE THREAD

**Issue by slugbyte** [slugbyte/lackluster.nvim#72]
while we're at it i think `FloatBorder` highlight also could be something better.
This is with default pallet:

![img](https://cdn.discordapp.com/attachments/976958356872908800/1282658413129764875/image.png?ex=66e0283c&is=66ded6bc&hm=f420e81124ff0341c1390cf74238e3c14b7a28deb9dfa72770a5fc839526fd2e&)

This is with `Floatborder` `guibg=#101010`:

![img](https://cdn.discordapp.com/attachments/976958356872908800/1282659759547351142/image.png?ex=66e0297d&is=66ded7fd&hm=bc627e47aadbcc20ecd30aafbe848a0698b151dd5c2d1fab7201a5d261b6f242&)

_Originally posted by @PsyNyde in https://github.com/slugbyte/lackluster.nvim/issues/69#issuecomment-2337828561_

**Comment by slugbyte** [comment]
as discussed in #69 this is approved,  I just need to figure out how I want to implement it :), I should have a patch in the next week :)

**Comment by slugbyte** [comment]
@PsyNyde Ive been messing around with this and haven't found a good solution

The main issue has to do with nvim-cmp.. basically when you tell nvim cmp to  use a boarders window it no longer use Pmenu as its background, it instead uses Normal as its background... 

this means that if I make it look good for nvim-cmp it will just end up not looking good for any other plugin that uses a float boarder

what nvim-cmp should have done is use NormalFloat, or Pmenu for the background and FloatBoarder 

so I'm not sure what I want to do going forward, I think adding a section in the docs on there readme to tellpeople how to have nice floatboarder for nvim-cmp might be the best option

```lua 
-- make nvim-cmp's boarders look pretty :) 
local lackluster = require("lackluster")
lackluster.setup({
    tweak_highlight = {
        FloatBorder = {
            fg = lackluster.color.gray5,
            bg = lackluster.color_specail.main_background,
        },
    },
})
vim.cmd.colorscheme("lackluster-hack")
```

**Comment by slugbyte** [comment]
https://github.com/hrsh7th/nvim-cmp/blob/ae644feb7b67bf1ce4260c231d1d4300b19c6f30/lua/cmp/config/window.lua#L7

^ this is the code in nvim-cmp that I think is non-standard 

I think it should start with `NormalFloat:NormalFloat` instead of `Normal:Normal`

**Comment by slugbyte** [comment]
I opened an issue with nvim-cmp https://github.com/hrsh7th/nvim-cmp/issues/2039 to see if we can solve this issue by patching nvim-cmp :)

**Comment by PsyNyde** [comment]
> So I'm not sure what I want to do going forward, I think adding a section in the docs on there readme to tell people how to have nice floatboarder for nvim-cmp might be the best option.

I also think this'd be the better option.

**Comment by slugbyte** [comment]
yep yep, `vim.cmd("highlight FloatBorder guibg=#101010")` looks good for nvim cmp but breaks other plugins that use `NormalFloat` for their backgound color including the example you gave example (which-key).

**Comment by slugbyte** [comment]
I'll wait for a reply on the nvim-cmp issue before I decide how to resolve this issue :)

**Comment by slugbyte** [comment]
created a new nvm-cmp issue after realizing the one i previously made did not properly use their bug template https://github.com/hrsh7th/nvim-cmp/issues/2042

**Comment by slugbyte** [comment]
@PsyNyde here is a snippet that you can use to test the proposed change to nvim-cmp if you are intersted 

```lua 
        cmp.setup({
            window = {
                completion = vim.tbl_extend("force", cmp.config.window.bordered(), {
                    winhighlight = "NormalFloat:NormalFloat,FloatBorder:FloatBorder,CursorLine:Visual,Search:None",
                }),
                documentation = vim.tbl_extend("force", cmp.config.window.bordered(), {
                    winhighlight = "NormalFloat:NormalFloat,FloatBorder:FloatBorder,CursorLine:Visual,Search:None",
                }),
            },
           -- rest of your config
     })
```

**Comment by PsyNyde** [comment]
> @PsyNyde here is a snippet that you can use to test the proposed change to nvim-cmp if you are intersted

hm this seems to fix the issue
![image](https://github.com/user-attachments/assets/86cca7d6-2cc4-45df-a161-f9fc0956cb12)

**Comment by fhawk12** [comment]
this fix my problem, maybe it works for you, too
```lua
		cmp.setup({
			snippet = {
				expand = function(args)
					vim.snippet.expand(args.body)
				end,
			},
			window = {
				completion = cmp.config.window.bordered(),
				documentation = cmp.config.window.bordered(),
			},
                        ...
    }
```

```lua
			lackluster.setup({
				tweak_syntax = {
					comment = lackluster.color.gray5,
				},
				tweak_background = {
					-- normal = "default", -- main background
					normal = "none", -- transparent
				},
				tweak_highlight = {
					FloatBorder = {
                                                overwrite = true,
						fg = lackluster.color.gray5,
						bg = "NONE",
					},
					["@keyword"] = {
						overwrite = true,
						bold = false,
						italic = true,
						fg = lackluster.color.gray6,
					},
					["@keyword.return"] = {
						overwrite = true,
						bold = false,
						italic = true,
						fg = lackluster.color.green,
					},
				},
			})

```
![image](https://github.com/user-attachments/assets/7dfd920b-f288-4e43-b53a-fea5d3fae2c3)

**Comment by slugbyte** [comment]
@fhawk12 thanks for sharing! Makes me realize I need to make a tweak to allow for transparency with floats.. I'm on holiday rn but I'll make a patch next week sometime! `bg_float = "none"` or something 

Funny enough though, it's one of those things we're the right outcome is happening but for the wrong reason, nvim-cmps bug essentially doesn't effect you because you wanted transparency, and the thing they are assigning to background for the cmp-window happens to be transparent, which is lucky I guess :) when/if the bug gets patched, it should still look the way you want it to but for the right reason :)

**Comment by slugbyte** [comment]
I got the changed merged into magazine.nvim (a nvim-cmp fork)

https://github.com/iguanacucumber/magazine.nvim/issues/3

**Comment by slugbyte** [comment]
Sorry, i have recently been way to busy with life and am not able to keep working on this project. If you want to submit a pr with a bug fix I would be happy to merge it. closing this issue for now.
